server/games/game.py

```diff
@@ -764,8 +774,8 @@ def get_player_outcome(self, player):
 
         return self._results.outcome(army)
 
-    def report_army_stats(self, stats):
-        self._army_stats = stats
+    def report_army_stats(self, stats_json):
+        self._army_stats_list = json.loads(stats_json)["stats"]
         self._process_pending_army_stats()
```

Comment:
I am hoping that this isn't called to often, so the json parsing doesn't happen too often, ideally only once at game end (or at least only once per player).
Let me know if that's not the case, then this should probably be saved as string and only parsed on-demand.

Reply:
I think `report_army_stats` is called from `GameConnection` each time a player sends army stats. I'm also pretty sure that army stats are reported in a similar way to game results where every player reports stats for every other player. The  difference is it might happen only at game end.

Reply:
As long as it's only called at the game end I think that's fine? I just didn't want to do unnecessary parsing of partial stats every minute if they are going to be overwritten anyways.

The parsing needed to be moved so that stats can be included in the `gameResults` message.